protocols/gossipsub/src/protocol.rs

```diff
@@ -246,6 +261,24 @@ impl Decoder for GossipsubCodec {
         let mut invalid_messages = Vec::new();
 
         for message in rpc.publish.into_iter() {
+            let topic = TopicHash::from_raw(&message.topic);
+
+            // Check the message size to ensure it doesn't bypass the configured max.
+            if message.get_size() > self.config.max_transmit_size_for_topic(&topic) {
+                let message = RawMessage {
+                    source: None, // don't bother inform the application
+                    data: message.data.unwrap_or_default(),
+                    sequence_number: None, // don't inform the application
+                    topic: TopicHash::from_raw(message.topic),
+                    signature: None, // don't inform the application
+                    key: message.key,
+                    validated: false,
+                };
+
+                invalid_messages.push((message, ValidationError::TransformFailed));
```

Comment:
Maybe we create a new errror. `MessageSizeTooLargeForTopic` to help identify these.